snapcraft_legacy/cli/_options.py

```diff
@@ -354,27 +363,28 @@ def get_build_provider_flags(build_provider: str, **kwargs) -> Dict[str, str]:
         raise RuntimeError(f"Invalid build provider: {build_provider}")
 
     for option in _PROVIDER_OPTIONS:
-        key: str = option["param_decls"]  # type: ignore
+        keys: List[str] = option["param_decls"]  # type: ignore
         is_flag: bool = option.get("is_flag", False)  # type: ignore
         envvar: Optional[str] = option.get("envvar")  # type: ignore
         supported_providers: List[str] = option["supported_providers"]  # type: ignore
 
-        # Skip --provider option.
-        if key == "--provider":
-            continue
+        for key in keys:
+            # Skip --provider option.
+            if key == "--provider":
+                continue
 
-        # Skip options that do not apply to configured provider.
-        if build_provider not in supported_providers:
-            continue
+            # Skip options that do not apply to configured provider.
+            if build_provider not in supported_providers:
+                continue
 
-        # Skip boolean flags that have not been set.
-        key_formatted = _param_decls_to_kwarg(key)
-        if is_flag and not kwargs.get(key_formatted):
-            continue
+            # Skip boolean flags that have not been set.
+            key_formatted = _param_decls_to_kwarg(key)
+            if is_flag and not kwargs.get(key_formatted):
+                continue
 
-        # Add build provider flag using envvar as key.
-        if envvar is not None and key_formatted in kwargs:
-            build_provider_flags[envvar] = kwargs[key_formatted]
+            # Add build provider flag using envvar as key.
+            if envvar is not None and key_formatted in kwargs:
+                build_provider_flags[envvar] = kwargs[key_formatted]
```

Comment:
This new code has a side effect of adding both `v` and `verbose` to the `build_provider_flags`.  I think this is harmless, but the proper solution would be to discard any single character flags (keep `verbose` and discard `v`).  I can add this on Monday.